### PR TITLE
Only run custom device validation code when needed

### PR DESCRIPTION
**Pull request by buckd**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/niveristand-routing-and-faulting-custom-device/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Prevents validation from running every time the page is changed in the Routing and Faulting custom device. This serves two purposes:

1. Page load performance improvements.
2. Don't create a dirty dot on the project each time the page is changed.

### Why should this Pull Request be merged?

There is no reason to re-run validation on every page change and we certainly should not be marking a project dirty if nothing has changed.

### What testing has been done?

Manual exercising of the sysdef when things changed.
Ran automated tests.

****Note:** Many automated tests are currently failing because the system definitions are not actually valid. Now that the Routing and Faulting custom device cares that the modules exist and are valid, this is a problem. I am working on updating the tests/sysdefs.

### CODE REVIEW

File: Source/Routing and Faulting/Scripting/Tests/Unit/Routing and Faulting Validation Unit Tests/Assets/Validation Test.nivssdf [hunk 1]
```diff
@@ -5665,6 +5677,24 @@
                      <Property Name="user.CD.Modules.5">
 								<DependentNode Path="Targets/Controller/Hardware/SLSC/SLSC Chassis/Modules/Unused" />
 							</Property>
+                     <Property Name="user.CD.Revisions.0">
```
Comment (rtzoeller) [inline Source/Routing and Faulting/Scripting/Tests/Unit/Routing and Faulting Validation Unit Tests/Assets/Validation Test.nivssdf]: There something strange happening with indentation here. I think you have a mix of tabs and spaces.

**Comment by rtzoeller** [comment]
- [ ] Instead of pushing the requirement to call `Update Configuration Revision.vi` onto the hardware custom device, the Routing and Faulting Hardware API should care of this automatically. Call `Update Configuration Revision.vi` from `Set All Possible Connections.vi`, and revert the changes to the SLSC Switch VIs.

File: Source/Routing and Faulting/Scripting/Tests/Unit/Routing and Faulting Validation Unit Tests/Assets/Validation Test.nivssdf (hunk 1 above)
Comment (buckd) [reply Source/Routing and Faulting/Scripting/Tests/Unit/Routing and Faulting Validation Unit Tests/Assets/Validation Test.nivssdf]: Fixed.

**Comment by csjall** [comment]
- [ ] `Set Validated Revision.vi`, `Get Module Configuration Revision from Identifier.vi`: Use consistent naming for getter/setter. This symmetry improves the readability/usability of the API. 
image
image

**Pull request merged**